packages/govuk-frontend/src/govuk/components/task-list/template.test.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * @jest-environment jsdom
+ */
+/* eslint-env jest */
+
+const { render } = require('govuk-frontend-helpers/nunjucks')
+const { getExamples } = require('govuk-frontend-lib/files')
+
+describe('Task List', () => {
+  let examples
+
+  beforeAll(async () => {
+    examples = await getExamples('task-list')
+  })
+
+  describe('default example', () => {
+    it('renders the default example', () => {
+      const $ = render('task-list', examples.default)
+
+      const $component = $('.govuk-task-list')
+      expect($component.get(0).tagName).toEqual('ul')
+
+      const $items = $component.find('.govuk-task-list__item')
+      expect($items.length).toEqual(3)
+
+      expect($items.get(0).tagName).toEqual('li')
+      expect($items.hasClass('govuk-task-list__item--with-link')).toBeTruthy()
+    })
+
+    it('associates the task name link with the status using aria', async () => {
+      const $ = render('task-list', examples.default)
+
+      const $itemLink = $('.govuk-task-list__link')
+      expect($itemLink.get(0).tagName).toEqual('a')
+      expect($itemLink.attr('href')).toEqual('#')
+
+      expect($itemLink.attr('aria-describedby')).toEqual('task-list-example-1-status')
+
+      const $statusWithId = $('#' + $itemLink.attr('aria-describedby'))
```

Comment:
Just looking into this test failure for you

We have type checking enabled and Cheerio expects selector strings to match:

```js
type SelectorType = `${SelectorSpecial}${AlphaNumeric}${string}` | `${AlphaNumeric}${string}`;
```

Otherwise it can't infer the element type and thinks `.tagName` might not exist 🤦‍♂️

Give this a try? It keeps the selector as a string but still checks `aria-describedby` matches

```js
const statusId = 'task-list-example-1-status'
expect($itemLink.attr('aria-describedby')).toEqual(statusId)

const $statusWithId = $(`#${statusId}`)
expect($statusWithId.get(0).tagName).toEqual('div')
```

Reply:
You can run the type checks locally with:

```console
npm run build:types
```

Reply:
@colinrotherham that didn't fix it, I got the same error. But changing from ``$(`#${statusId}`)`` to ``$component.find(`#${statusId}`)`` does. I have no idea why. 🤷